### ISSUE TITLE
PyTorch1.2 ONNX dynamic_axis is not working

### ISSUE THREAD

**Issue by Mut1nyJD** [pytorch/pytorch#27695]
## 🐛 Bug

When I am using ONNX export with dynamic axis I'll always get a warning from inside utils.py in torch/onnx saying that the input or output name can not be found which is not true. After having inspected the code itself it is clear that adding them to the valid_name set is simply wrong as it is adding an expression instead of the execution of the expression into the set.

## To Reproduce

Simple call something along the lines of this:

You can use any model that expects a 4-dim input tensor

torch.onnx.export(pytorch_model,tuple((input,)),'test.onnx', verbose=True,input_names=['input'],output_names=['output'],dynamic_axes={'input':[0,2,3],'output':[0,2,3]})

## Environment
PyTorch 1.2.0
torchvision 0.4.0
Python 3.7.3
Ubuntu Linux 19.xx

**Comment by BowenBao** [comment]
Hi @Mut1nyJD, based on your description I have created an example model with 4-dim input tensor, but I could not repro the issue.
```python
import torch
class Test(torch.nn.Module):
    def forward(self, x):
        return x + 1

x = torch.ones(2,3,4,5)
torch.onnx.export(Test(), tuple((x,)), 'test.onnx', verbose=True,
                  input_names=['input'], output_names=['output'],
                  dynamic_axes={'input':[0, 2, 3], 'output':[0, 2, 3]})
```
Warnings are legit, mentioning that the symbolic names for axes are generated since not provided.
```
utils.py:780: UserWarning: No names were found for specified dynamic axes of provided input.Automatically generated names will be applied to each dynamic axes of input input
  'Automatically generated names will be applied to each dynamic axes of input {}'.format(key))
utils.py:780: UserWarning: No names were found for specified dynamic axes of provided input.Automatically generated names will be applied to each dynamic axes of input output
  'Automatically generated names will be applied to each dynamic axes of input {}'.format(key))
```
Exported ONNX graph:
![Screenshot from 2019-11-05 11-14-33](https://user-images.githubusercontent.com/9376104/68238174-7ba7b100-ffbd-11e9-86d1-3144ee2ad65b.png)

**Comment by spandantiwari** [comment]
@Mut1nyJD - please reopen if there's still any concern.

**Comment by gbmarc1** [comment]
I agree with @Mut1nyJD,


    valid_names = set()
    if input_names is not None:
        valid_names.add(x for x in input_names)
    if output_names is not None:
        valid_names.add(x for x in output_names)

    # If dynamic axes are provided as a list rather than dictionary, they should
    # first get converted to a dictionary in expected format. If desired axes names
    # are not provided for dynamic axes, automatic names shall be generated for
    # provided dynamic axes of specified input/output
    for key, value in dynamic_axes.items():
        if key not in valid_names:
            warnings.warn("Provided key {} for dynamic axes is not a valid input/output name".format(key))


`x for x in output_names` is a generator
In `if key not in valid_names` you are comparing a generator with a string. I don't think you will ever get a match.

**Comment by BowenBao** [comment]
@gbmarc1 Thanks for adding more info.
Please use newer version PyTorch. This issue is already resolved.
https://github.com/pytorch/pytorch/blob/8a2dcff189b7b20e1d247f36ee89b902e5be5a88/torch/onnx/utils.py#L763
